### ISSUE TITLE
amtool: amtool config and alert issues.

### ISSUE THREAD

**Issue by mjtrangoni** [prometheus/alertmanager#746]
Hi,

I compiled today the latest release v0.6.0 with go1.8.1 and I wanted to report the issues I am facing with `alert` and `config` commands. Just `silence` is working as expected. See,

```
$ ldd /sbin/amtool
        linux-vdso.so.1 =>  (0x00007ffeabbf9000)
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007f5b819d9000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f5b81618000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f5b81c01000)
$ /sbin/amtool config
panic: json: cannot unmarshal object into Go struct field Route.match_re of type string

goroutine 1 [running]:
github.com/prometheus/alertmanager/cli.fetchConfig(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/cli/config.go:81 +0x4f1
github.com/prometheus/alertmanager/cli.queryConfig(0xbfa2a0, 0xc1dcd8, 0x0, 0x0, 0x0, 0x0)
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/cli/config.go:93 +0x37
github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra.(*Command).execute(0xbfa2a0, 0xc1dcd8, 0x0, 0x0, 0xbfa2a0, 0xc1dcd8)
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra/command.go:632 +0x3ef
github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xbfa4c0, 0x0, 0x975b25, 0x27)
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra/command.go:722 +0x339
github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra.(*Command).Execute(0xbfa4c0, 0x982cb8, 0xc42013cc90)
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/vendor/github.com/spf13/cobra/command.go:681 +0x2b
github.com/prometheus/alertmanager/cli.Execute()
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/cli/root.go:40 +0x2d
main.main()
        /scr/sargas/trangoni/go/packages/src/github.com/prometheus/alertmanager/cmd/amtool/main.go:6 +0x20
$ /sbin/amtool alert
Error: Unable to decode json response
```

**Comment by stuartnelson3** [comment]
Are you also using the latest version of alertmanager on the server?

**Comment by mjtrangoni** [comment]
Hi @stuartnelson3,

We are running the latest releases, see
```
# uname -a
Linux hostname 3.10.0-514.16.1.el7.x86_64 #1 SMP Wed Apr 12 15:04:24 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
# /sbin/alertmanager --version
alertmanager, version 0.6.1 (branch: HEAD, revision: 5aeaf2cb988b0094b7998bbe6568bba23a0eb9c2)
  build user:       trangoni@sargas
  build date:       20170428-12:14:06
  go version:       go1.8.1
# /sbin/prometheus --version
prometheus, version 1.6.0 (branch: HEAD, revision: 10f645382988add09c8e3420d0d7ff2dc4d36a29)
  build user:       trangoni@sargas
  build date:       20170418-10:40:17
  go version:       go1.8.1
```

We see that it has something with parsing `match_re`, that at the end results in an empty service,

```
  - match_re:
      service: '^(foo1|foo2|baz)$'
    receiver: prometheus-mails
    # The service has a sub-route for critical alerts, any alerts
    # that do not match, i.e. severity != critical, fall-back to the
    # parent node and are sent to 'team-X-mails'
    routes:
    - match:
        severity: critical
      receiver: team-X-pager
```

Any advice? The configuration is almost the default one, aside of emails.

**Comment by Kellel** [comment]
Oh I see. I'm able to reproduce. I'll see whats up.

**Comment by Kellel** [comment]
@stuartnelson3 you can go ahead and assign this to me :smile:

**Comment by stuartnelson3** [comment]
Hah cool, thanks! Thought we had dealt with this, but 🤷‍♂️

**Comment by stuartnelson3** [comment]
Seems I can't assign you, but if anyone is looking, this issue is being addressed

**Comment by Kellel** [comment]
Hmm looks like alertmanager is struggling with the config. Bug isn't actually in amtool. (I'll still try and fix it :smile: )
`kellen@chewbacca:~/.GOPATH/src/github.com/prometheus/alertmanager$ curl -s http://localhost:9093/api/v1/status | jq .data.configJSON.route.routes[0]
{
  "receiver": "team-X-mails",
  "match_re": {
    "service": {}
  },
  "routes": [
    {
      "receiver": "team-X-pager",
      "match": {
        "severity": "critical"
      }
    }
  ]
}
`

**Comment by fabxc** [comment]
I think that issue is fixed as part of #773
